File: dietideals-service/src/main/java/it/uninastudents/dietidealsservice/model/dto/UtenteRegistrazione.java
package it.uninastudents.dietidealsservice.model.dto;

import it.uninastudents.dietidealsservice.model.entity.enums.RuoloUtente;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Past;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@EqualsAndHashCode
@NoArgsConstructor
public class UtenteRegistrazione {

    @NotBlank
    private String username;

    @Enumerated(value = EnumType.STRING)
    private RuoloUtente ruolo;

    @NotBlank
    private String nome;

    @NotBlank
    private String cognome;

    @Past
    private LocalDate dataNascita;

    @Email
    private String email;

    private String password;

    @Pattern(regexp = "^\\d{11}$|^$")
    private String partitaIva;

    private String urlDocumentoIdentita;

    private String urlFotoProfilo;

    @Valid
    private CreaContoCorrente contoCorrente;

}

File: dietidealsapp/app/src/main/java/com/example/dietideals_app/model/dto/UtenteRegistrazione.kt
package com.example.dietideals_app.model.dto

import com.example.dietideals_app.model.ContoCorrente
import com.example.dietideals_app.model.enum.RuoloUtente

class UtenteRegistrazione(
    val username: String,
    val ruolo: RuoloUtente,
    val nome: String,
    val cognome: String,
    var dataNascita: String,
    val email: String,
    val password: String?,
    val partitaIva: String?,
    val urlDocumentoIdentita: String?,
    val urlFotoProfilo: String?,
    val contoCorrente: ContoCorrente?
) {
}